lua/genghis/init.lua

```diff
@@ -160,19 +160,50 @@ function M.chmodx()
 	cmd.edit()
 end
 
----@param opts? {trashLocation: string}
+---@param opts? {trashCmd: string, trashLocation: string}
 function M.trashFile(opts)
 	cmd.update { bang = true }
 	local home = os.getenv("HOME")
 	local oldFilePath = vim.api.nvim_buf_get_name(0)
 	local oldName = vim.fs.basename(oldFilePath)
 
+	local trashCmd
+	if opts ~= nil and opts.trashCmd ~= nil then
+		trashCmd = opts.trashCmd
+	else
+		if fn.has("linux") == 1 then
+			trashCmd = "gio trash"
+		elseif fn.has("windows") == 1 then
+			trashCmd = "trash"
+		end
+	end
+
+	if trashCmd ~= nil then
+		-- Use a trash command
+		local trashArgs = vim.split(trashCmd, " ")
+		table.insert(trashArgs, oldFilePath)
+
+		local errMsg = ""
+		vim.fn.jobstart(trashArgs, {
+			detach = true,
+			on_stderr = function (_, data)
+				errMsg = errMsg .. (data and table.concat(data, " "))
+			end,
+			on_exit = function(_, rc)
+				if rc == 0 then
+					u.bwipeout()
+					u.notify(("%q deleted"):format(oldName))
+				else
+					u.notify(("Trashing %q failed: " .. errMsg):format(oldName), "warn")
```

Comment:
on failure, the severity level should be "error", not "warning" (since the operation did not take place at all)